sale_order_import/wizard/sale_order_import.py

```diff
@@ -0,0 +1,663 @@
+        assert filename, "Missing filename"
+        assert filecontent, "Missing file content"
+        filetype = mimetypes.guess_type(filename)
+        logger.debug("Order file mimetype: %s", filetype)
+        mimetype = filetype[0]
+        supported_types = {
+            "CSV": ("text/csv", "text/plain"),
+            "XML": ("application/xml", "text/xml"),
+            "PDF": ("application/pdf"),
+        }
+        res = None
+        if filetype and mimetype in supported_types["CSV"]:
+            res = (
+                self.parse_csv_order(filecontent, self.partner_id)
+                if self.partner_id
+                else False
+            )
+        elif filetype and mimetype in supported_types["XML"]:
+            xml_root, error_msg = self._parse_xml(filecontent)
+            if (xml_root is None or not len(xml_root)) and error_msg:
+                raise UserError(error_msg)
+            res = self.parse_xml_order(xml_root, detect_doc_type=detect_doc_type)
+        elif filetype and mimetype == supported_types["PDF"]:
+            res = self.parse_pdf_order(filecontent, detect_doc_type=detect_doc_type)
+        return res
+
+    def _unsupported_file_msg(self, filename):
+        return {
+            "title": _("Unsupported file format"),
+            "message": _(
+                "This file '%s' is not recognised as a CSV, XML nor "
+                "PDF file. Please check the file and it's "
+                "extension."
+            )
+            % filename,
+        }
+
+    @api.model
+    def _parse_xml(self, data):
+        if not data:
+            return None, _("No data provided")
+        xml_root = None
+        try:
+            xml_root = etree.fromstring(data)
+            error_msg = None
+        except etree.XMLSyntaxError:
+            error_msg = _("This XML file is not XML-compliant")
+            return xml_root, error_msg
+        try:
+            self.parse_xml_order(xml_root, detect_doc_type=True)
+        except (UserError, NotImplementedError):
+            error_msg = _("Unsupported XML document")
+        return xml_root, error_msg
+
+    # FIXME: not used at all
+    @api.model
+    def get_xml_doc_type(self, xml_root):  # pragma: no cover
+        raise UserError
+
+    @api.model
+    def parse_xml_order(self, xml_root, detect_doc_type=False):
+        raise NotImplementedError(
+            _(
+                "This type of XML RFQ/order is not supported. Did you install "
+                "the module to support this XML format?"
+            )
+        )
+
+    # FIXME: not used at all
+    @api.model
+    def parse_csv_order(self, order_file, partner):  # pragma: no cover
+        assert partner, "missing partner"
+        raise UserError(
+            _(
+                "This type of CSV order is not supported. Did you install "
+                "the module to support CSV orders?"
+            )
+        )
+
+    # TODO: move it out to a PDF support module
```

Comment:
this is a new import type  in theory. Yet, as is still based on xml parsing in the end, we can leave it here but refactored according to the "import_type" mentioned above